### PR TITLE
External L4B support

### PR DESCRIPTION
**Pull request by YakhontovYaroslav**
This PR consists of changes made by @borkmann in https://github.com/cilium/cilium/pull/31163, adapted to current master and unlocks use case described [here](https://github.com/cilium/cilium/issues/33611).

This is resulting traffic flow, once this change gets merged
image

### CODE REVIEW

**Review by lmb — APPROVED** [review]
Loader changes LGTM.

File: pkg/option/config.go [hunk 1]
```diff
@@ -1689,6 +1694,11 @@ type DaemonConfig struct {
 	// WireguardPersistentKeepalive controls Wireguard PersistentKeepalive option.
 	WireguardPersistentKeepalive time.Duration
 
+	// EnableExternalDSR is the name of the option to terminate connections from
+	// an external L4LB in front of the K8s cluster, and to DSR replies back to
+	// the client from the K8s cluster.
+	EnableExternalDSR bool
```
Comment (joamaki) [inline pkg/option/config.go:1700]: Should there be an helm value for this or how would this be configured?

**Review by joamaki — CHANGES_REQUESTED** [review]
The commit history needs to be cleaned up (there should be no merge commits) and a proper descriptive commit message is needed (e.g. the original commit messages should be kept). It seems like the helm changes are missing as well. And it probably would make sense to document this in https://docs.cilium.io/en/stable/network/kubernetes/kubeproxy-free/#dsr-mode? e2e tests would also be important to implement as otherwise this will for sure break in the future, or at least BPF unit tests (`bpf/tests`).

**Comment by tommyp1ckles** [comment]
Thanks for the change 😃 

CLI changes look good to me, just waiting on the commit history.  For more information regarding what @joamaki was saying you can take a look at [this section of the docs](https://docs.cilium.io/en/latest/contributing/development/contributing_guide/#submitting-a-pull-request).

**Comment by joestringer** [comment]
Marking this PR as draft while the feedback is addressed. Once you think this is ready for further review, you can click the "Ready for Review" button at the bottom of the page. Thank you for your contributions.

**Pull request closed without merging**